src/bokeh/core/property/wrappers.py

```diff
@@ -468,10 +470,26 @@ def _stream(self, doc: Document, source: ColumnarDataSource, new_data: dict[str,
         # is actually the already updated value. This is because the method
         # self._saved_copy() makes a shallow copy.
         for k in new_data:
-            if isinstance(self[k], np.ndarray) or isinstance(new_data[k], np.ndarray):
-                data = np.append(self[k], new_data[k])
+            old_array = cast(Sequence[Any], self[k])
+            new_array = cast(Sequence[Any], new_data[k])
```

Comment:
Can these casts be moved below to where `data` is actually assigned? 

Also a this point it is not know that the values are arrays, suggest `old_data` and `new_data` as better names.

Reply:
I improved the typing situation, so that casts aren't needed. However, the point of the assignment is to avoid indexing multiple times in this code. I renamed `*_array` to `*_seq`, because `new_data` is already used.